### ISSUE TITLE
Password Reset Framework Breaks When Altering Where Email Is Stored

### ISSUE THREAD

**Issue by justautomated** [laravel/framework#14081]
**Background**
I ran the default authentication/registration scaffolding [php artisan make:auth].  I needed to alter authenticating with a username/password rather than email/password so I converted the model/views/config/tables and I can authenticate and register.  That is working great.

**Issue**
The email of a user is located in another model/class (with a foreign key back to the user model).  Inside the User model I attempted to override the getEmailAttribute function like I've done to many of the other models but the password reset framework doesn't access the email attribute in this manner and when you attempt to submit your email address [http://localhost:8000/password/reset] laravel throws a SQL error: "Column not found: 1054 Unknown column 'email' in 'where clause' "

In a nutshell, my user's username/password are in the out-of-the-box User model but their email is in a related model/table.

**Comment by themsaid** [comment]
@justautomated the existing scaffold is written with a convenience in mind, for special cases like this one I suggest you build your own or try to extend the current one.

Closing since this is not actually a bug report.
